code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,275 @@
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("The King will hatch in approximately 5 minutes."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King will hatch in approximately 5 minutes."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+#define KING_PLAYTIME_HOURS (50 HOURS)
+
+/obj/effect/alien/resin/king_cocoon/proc/is_candidate_valid(datum/hive_status/hive, mob/candidate, playtime_restricted = TRUE, skip_playtime=TRUE)
+	if(!candidate.client)
+		return FALSE
+	if(isfacehugger(candidate) || islesserdrone(candidate))
+		return FALSE
+	if(playtime_restricted)
+		if(candidate.client.get_total_xeno_playtime() < KING_PLAYTIME_HOURS)
+			return FALSE
+	else if(candidate.client.get_total_xeno_playtime() >= KING_PLAYTIME_HOURS && skip_playtime)
+		return FALSE // We do this under the assumption we tried it the other way already so don't ask twice
+	for(var/mob_name in hive.banished_ckeys)
+		if(hive.banished_ckeys[mob_name] == candidate.ckey)
+			return FALSE
+	return TRUE
+
+/obj/effect/alien/resin/king_cocoon/proc/try_roll_candidate(datum/hive_status/hive, mob/candidate, playtime_restricted = TRUE)
+	if(!is_candidate_valid(hive, candidate, playtime_restricted))
+		return FALSE
+	
+	return tgui_alert(candidate, "Would you like to become the King?", "Choice", list("Yes", "No"), 10 SECONDS) == "Yes"
+
+#undef KING_PLAYTIME_HOURS
+
+/obj/effect/alien/resin/king_cocoon/proc/make_vote(mob/candidate, list/mob/living/carbon/xenomorph/voting_candidates)
+	var/mob/living/carbon/xenomorph/choice = tgui_input_list(candidate, "Vote for a sister you wish to become the King.", "Choose a xeno", voting_candidates , 20 SECONDS)
+
+	if(votes[choice])
+		votes[choice] += 1
+	else
+		votes[choice] = 1
+
+/obj/effect/alien/resin/king_cocoon/proc/roll_candidates()
+	var/datum/hive_status/hive = GLOB.hive_datum[hive_number]
+	
+	// First, let the living xenos choose
+	var/list/mob/living/carbon/xenomorph/voting_candidates = hive.totalXenos.Copy() - hive.living_xeno_queen
+
+	for(var/mob/living/carbon/xenomorph/voting_candidate in voting_candidates)
+		if(!is_candidate_valid(hive, voting_candidate))
+			voting_candidates -= voting_candidate
+
+	for(var/mob/living/carbon/xenomorph/candidate in hive.totalXenos)
+		if(is_candidate_valid(hive, candidate, FALSE, FALSE))
+			INVOKE_ASYNC(src, PROC_REF(make_vote), candidate, voting_candidates)
+	
+	sleep(20 SECONDS)
+	votes = sortAssoc(votes)
+	var/index = 0
+	for(var/mob/living/carbon/xenomorph/candidate in votes)
+		if(index == 0 && prob(50) && length(votes) > 1)
+			continue
+
+		if(index > 1)
+			break
+
+		if(try_roll_candidate(hive, candidate, playtime_restricted = TRUE))
+			return candidate.client
+		
+		voting_candidates -= candidate
+
+		index++
+
+	// Ask all the living xenos
+	for(var/mob/living/carbon/xenomorph/candidate in voting_candidates)
```

Comment:
```suggestion
	for(var/mob/living/carbon/xenomorph/candidate in shuffle(voting_candidates))
```
Otherwise its age ordered. Which isn't awful but we've been shuffling before. So up to you if you'd rather this one not be shuffled.